src/app/core/classes/system-profiler.ts

```diff
@@ -182,9 +182,7 @@ export class SystemProfiler {
         v.disks[name] = -1; // no children so we use this as placeholder
       } else if (vdev.children.length > 0) {
         vdev.children.forEach((disk, dIndex) => {
-          if (!disk.device && disk.status == 'REMOVED') {
-
-          } else {
+          if (disk.device && disk.status != 'REMOVED') {
```

Comment:
The new condition would be true for a modal
`disk.device = true;
disk.status = 'REMOVED'`

The condition isn't exactly matched. With these values, the code would run before in the else block but won't now